### ISSUE TITLE
Retry logic

### ISSUE THREAD

**Issue by ened** [BlueChilli/flutter_uploader#1]
How does this plugin handle failed uploads due to connectivity issues? From my understand uploads are currently marked as failed and will not resume?

**Comment by onikiri2007** [comment]
Yes that's correct, if upload failed due to connectivity, it will raise failed event.

**Comment by ened** [comment]
Could this be adjusted with an exponential retry logic? Some files really must be uploaded - and the user will not want to re-upload manually if their internet connection is flaky.

**Comment by onikiri2007** [comment]
in order to add above feature,  upload probably requires some kind of flag or retry policy to be as parameters so uploader can decide whether to retry exponentially or stop at a certain number of failures. we can also probably introduce like downloader plugin to record and track upload task in the db.

**Comment by ened** [comment]
OK. I will get to it at some point. Currently extending this plugin with support for raw uploads (not only multipart/form-data) and few other bits.

**Comment by onikiri2007** [comment]
Thats great looking forward to your pull  request :grinning:

**Comment by Iradukunda-Nadia** [comment]
@ened was this ever implemented?
